crates/uv-installer/src/site_packages.rs

```diff
@@ -319,6 +320,22 @@ impl SitePackages {
                             RequirementSatisfaction::Satisfied => {}
                         }
                     }
+                    // Validate that the installed version satisfies the overrides.
```

Comment:
I think we'd need to change the code such that all _dependencies_ also respect overrides.

Reply:
Yeah I noticed this in my test at https://github.com/astral-sh/uv/pull/4973/files#diff-a7df5406f8301d2dd603b5355f282be5afa0198ff71f03144ab6d8aed56c07a3R713-R719 — I'll explore fixing this before merging here. I'm not sure how I'll write a nice test case though without the `uv run` changes.